app/screens/invite/selection.tsx

```diff
@@ -0,0 +1,392 @@
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import SelectedUser from '@components/selected_users/selected_user';
+import TouchableWithFeedback from '@components/touchable_with_feedback';
+import UserItem from '@components/user_item';
+import {MAX_LIST_HEIGHT, MAX_LIST_TABLET_DIFF} from '@constants/autocomplete';
+import {useTheme} from '@context/theme';
+import {useAutocompleteDefaultAnimatedValues} from '@hooks/autocomplete';
+import {useIsTablet, useKeyboardHeight} from '@hooks/device';
+import {makeStyleSheetFromTheme, changeOpacity} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+import {SearchResult} from './invite';
+import SelectedEmail from './selected_email';
+import SelectionSearchBar from './selection_search_bar';
+import SelectionTeamBar from './selection_team_bar';
+import TextItem, {TextItemType} from './text_item';
+
+const AUTOCOMPLETE_ADJUST = 5;
+const KEYBOARD_HEIGHT_ADJUST = 3;
+
+const INITIAL_BATCH_TO_RENDER = 15;
+const SCROLL_EVENT_THROTTLE = 60;
+
+const keyboardDismissProp = Platform.select({
+    android: {
+        onScrollBeginDrag: Keyboard.dismiss,
+    },
+    ios: {
+        keyboardDismissMode: 'on-drag' as const,
+    },
+});
+
+const keyExtractor = (item: SearchResult) => (
+    typeof item === 'string' ? item : (item as UserProfile).id
+);
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            display: 'flex',
+            flex: 1,
+        },
+        teamContainer: {
+            display: 'flex',
+            flexDirection: 'row',
+            alignItems: 'center',
+            width: '100%',
+            paddingVertical: 16,
+            paddingHorizontal: 20,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.04),
+        },
+        iconContainer: {
+            width: 40,
+            height: 40,
+        },
+        textContainer: {
+            display: 'flex',
+            flexDirection: 'column',
+        },
+        teamText: {
+            color: theme.centerChannelColor,
+            marginLeft: 12,
+            ...typography('Body', 200, 'SemiBold'),
+        },
+        serverText: {
+            color: changeOpacity(theme.centerChannelColor, 0.72),
+            marginLeft: 12,
+            ...typography('Body', 75, 'Regular'),
+        },
+        shareLink: {
+            display: 'flex',
+            marginLeft: 'auto',
+        },
+        shareLinkButton: {
+            display: 'flex',
+            flexDirection: 'row',
+            alignItems: 'center',
+            height: 40,
+            paddingHorizontal: 20,
+            backgroundColor: changeOpacity(theme.buttonBg, 0.08),
+            borderRadius: 4,
+        },
+        shareLinkText: {
+            color: theme.buttonBg,
+            ...typography('Body', 100, 'SemiBold'),
+            paddingLeft: 7,
+        },
+        shareLinkIcon: {
+            color: theme.buttonBg,
+        },
```

Comment:
These aren't used in the file.

Reply:
Remained after splitting the initial component, removed.